engine/org.eclipse.birt.report.engine.emitter.wpml/src/org/eclipse/birt/report/engine/emitter/wpml/EmitterServices.java

```diff
@@ -0,0 +1,418 @@
+		}
+
+		if ((value == null) && (options != null)) {
+			value = options.getOption(name);
+		}
+
+		if (reportContent != null && value == null) {
+			value = getReportDesignConfiguration(reportContent, name);
+		}
+
+		if (value != null) {
+			result = booleanOption(value, defaultValue);
+		}
+
+		return result;
+	}
+
+	/**
+	 * Search for an emitter option and return it as a string
+	 *
+	 * @param options      The task options to extract the value from.
+	 * @param birtContent  The leaf node to look for UserProperties
+	 * @param name         The name of the value to extract from options.
+	 * @param defaultValue Value to return if value is null.
+	 * @return a string, or the defaultValue
+	 */
+	public static String stringOption(ITaskOption options, IContent birtContent, String name, String defaultValue) {
+		String result = defaultValue;
+		Object value = null;
+
+		IElement currentElement = birtContent;
+
+		while ((currentElement != null) && (value == null)) {
+			if (currentElement instanceof IContent) {
+				Object designObject = ((IContent) currentElement).getGenerateBy();
+				if (designObject instanceof ReportElementDesign) {
+					Map<String, Expression> userProperties = ((ReportElementDesign) designObject).getUserProperties();
+					if (userProperties != null) {
+						Expression expression = userProperties.get(name);
+						if (expression instanceof Expression.Constant) {
+							Expression.Constant constant = (Expression.Constant) expression;
+							value = constant.getValue();
+						}
+					}
+				}
+			}
+			if (value == null) {
+				currentElement = currentElement.getParent();
+			}
+		}
+		if ((value == null) && (birtContent != null)) {
+			Map<String, Expression> userProperties = birtContent.getReportContent().getDesign().getUserProperties();
+			if (userProperties != null) {
+				Expression expression = userProperties.get(name);
+				if (expression instanceof Expression.Constant) {
+					Expression.Constant constant = (Expression.Constant) expression;
+					value = constant.getValue();
+				}
+			}
+		}
+
+		if ((value == null) && (options != null)) {
+			value = options.getOption(name);
+		}
+
+		if (birtContent != null && birtContent.getReportContent() != null && value == null) {
+			value = getReportDesignConfiguration(birtContent.getReportContent(), name);
+		}
+
+		if (value != null) {
+			result = value.toString();
+		}
+
+		return result;
+	}
+
+	/**
+	 * Search for an emitter option and return it as a string
+	 *
+	 * @param options       The task options to extract the value from.
+	 * @param reportContent The report
+	 * @param name          The name of the value to extract from options.
+	 * @param defaultValue  Value to return if value is null.
+	 * @return a string, or the defaultValue
+	 */
+	public static String stringOption(ITaskOption options, IReportContent reportContent, String name,
+			String defaultValue) {
+		String result = defaultValue;
+		Object value = null;
+
+		if (reportContent != null) {
```

Comment:
This change cannot be done because there are cascading checks and in you situation we would loose to option of the check for "ITaskOption options" but this will be done afterwards.
If the value is again null then "getReportDesignConfiguration()" will be used for a last try.